src/components/pages/Map.tsx

```diff
@@ -92,6 +63,41 @@ function Map() {
   }, []);
   // </表示範囲判定>
 
+  // <infoWindowオプション-->
+  const [size, setSize] = useState<undefined | google.maps.Size>(undefined);
+  const infoWindowOptions = {
+    pixelOffset: size
+  };
+  const createOffsetSize = () => {
+    return setSize(new window.google.maps.Size(0, -45));
+  };
+
+  const [LatLng, setLatLng] = useState<google.maps.LatLng | google.maps.LatLngLiteral>();
+  // </infoWindowオプション-->
+
+  // <座標データ取得 未実装>
+  const [shops, setShops] = useState([]);
+  type shops = {
+    id: number;
+    name: string;
+    lat: number;
+    lng: number;
+    address :string
+    opening_hours :number
+    photo_reference :string
+    rating :number
+  };
+
+  // ↓Railsから表示範囲内のデータを取得
+  useEffect(() => {
+  axios.get(
+    'http://localhost:3001/api/v1/shops/?minlat='+swBounds?.lat+'&minlng='+swBounds?.lng+'&maxlat='+neBounds?.lat+'&maxlng='+neBounds?.lng)
+    .then(res => {setShops(res.data)})
+    .catch(error => console.log(error))
+  },[shops]);
```

Comment:
第二引数にshopsを渡すと無限ループ走っちゃいますね！
```suggestion
  }, []);
```

Reply:
無限ループ回避できました！
ありがとうございます！